explorer/templates.go

```diff
@@ -151,6 +151,42 @@ func float64Formatting(v float64, numPlaces int, useCommas bool, boldNumPlaces .
 	return []string{integer, dec[:places], dec[places:], trailingZeros}
 }
 
+func amountAsDecimalPartsTrimmed(v, numPlaces int64, useCommas bool) []string {
+	amt := strconv.FormatInt(v, 10)
+	intpart := v / 1e8
+	decpart := v % 1e8
+
+	if len(amt) <= 8 {
+		right := strings.TrimRight(amt, "0")
+		left := ""
+		tail := strings.Repeat("0", 8-len(amt))
+		if (8 - len(amt)) > 2 {
+			tail = "00"
+		}
+		return []string{"0", left + right, tail}
+	}
+
+	left := strconv.FormatInt(intpart, 10)
+	right := fmt.Sprintf("%08d", decpart)
+	right = string(right[0:numPlaces])
+
+	tail := ""
```

Comment:
???
Please comment the code more.

e.g. Everything before the `if useCommas` can be simplified and made more clear and robust:


```go
func amountAsDecimalPartsTrimmed(v, numPlaces int64, useCommas bool) []string {
	// Separate into int.dec parts.
	intpart := v / 1e8
	decpart := v % 1e8

	// Format left side.
	left := strconv.FormatInt(intpart, 10)

	// Format right side with 8 digits, including leading and trailing zeros.
	rightWithTail := fmt.Sprintf("%08d", decpart)
	// Reduce precision according to numPlaces.
	if len(rightWithTail) > int(numPlaces) {
		rightWithTail = rightWithTail[0:numPlaces]
	}

	// Separate trailing zeros.
	right := strings.TrimRight(rightWithTail, "0")
	tail := strings.TrimPrefix(rightWithTail, right) // Repeat("0", len(rightWithTail) - len(rightWithTail))
```

Reply:
173 no longer used